Bottom: 60de8e49cd40fa3fc5143b481f592c40ac9ad4dc
Top:    df890f27ffe28a15dca1df9621959de72ea0ab06
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-04 21:16:34 +0000

Play with grid snapping heuristics a little

Require the pointer to be inside the bounding box of a pin / pad / via before snapping.
This makse it MUCH easier to place vias in a regular grid spaced at the PCB grid spacing.


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 970db92..09a9fb9 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -967,7 +967,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PAD_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1024,7 +1024,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PIN_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1042,7 +1042,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (VIA_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid snapping vias to any other vias */
   if (Settings.Mode == MOVE_MODE &&
